opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+    token_expiry: Option<DateTime<Utc>>,
+}
+
+#[allow(dead_code)]
+pub(crate) struct GenevaConfigClient {
+    config: GenevaConfigClientConfig,
+    http_client: Client,
+    cached_data: RwLock<CachedAuthData>,
+}
+
+/// Client for interacting with the Geneva Configuration Service.
+///
+/// This client handles authentication and communication with the Geneva Config
+/// API to retrieve configuration information like ingestion endpoints.
+impl GenevaConfigClient {
+    /// Creates a new Geneva Config Client with the provided configuration.
+    ///
+    /// # Arguments
+    /// * `config` - The client configuration
+    ///
+    /// # Returns
+    /// * `Result<Self>` - A new client instance or an error
+    ///
+    /// # Errors
+    /// * `GenevaConfigClientError::Certificate` - If certificate authentication fails
+    /// * `GenevaConfigClientError::Io` - If reading certificate files fails
+    /// * `GenevaConfigClientError::Tls` - If TLS configuration fails
+    #[allow(dead_code)]
+    pub(crate) async fn new(config: GenevaConfigClientConfig) -> Result<Self> {
+        let mut client_builder = Client::builder()
+            .http1_only()
+            .timeout(Duration::from_secs(30))
+            .pool_idle_timeout(Duration::from_secs(90))
+            .pool_max_idle_per_host(16);
+
+        match &config.auth_method {
+            // TODO: Certificate auth would be removed in favor of managed identity.,
+            // This is for testing, so we can use self-signed certs, and password in plain text.
+            AuthMethod::Certificate { path, password } => {
+                // Read the PKCS#12 file
+                let p12_bytes = fs::read(path)?;
+                let identity = Identity::from_pkcs12(&p12_bytes, password)?;
+                let tls_connector =
+                    configure_tls_connector(native_tls::TlsConnector::builder(), identity)
+                        .build()?;
+                client_builder = client_builder.use_preconfigured_tls(tls_connector);
+            }
+            AuthMethod::ManagedIdentity => {
+                return Err(GenevaConfigClientError::Certificate(
+                    "Managed Identity authentication is not implemented yet".into(),
+                ));
+            }
+        }
+
+        let http_client = client_builder.build()?;
+
+        Ok(Self {
+            config,
+            http_client,
+            cached_data: RwLock::new(CachedAuthData {
+                auth_info: None,
+                token_endpoint: None,
+                token_expiry: None,
+            }),
+        })
+    }
+
+    fn parse_token_expiry(&self, expiry_str: &str) -> Option<DateTime<Utc>> {
+        // Attempt to parse the ISO 8601 datetime string
+        DateTime::parse_from_rfc3339(expiry_str)
+            .ok()
+            .map(|dt| dt.with_timezone(&Utc))
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn get_token_expiry(&self) -> Option<DateTime<Utc>> {
```

Comment:
Do we expect this method to be called?

Reply:
I anticipated this could be required when Geneva config fetch is added. But have removed it, and will bring it back at that time.